bundles/org.openhab.binding.folderwatcher/src/main/java/org/openhab/binding/folderwatcher/internal/handler/FolderLocalWatcherHandler.java

```diff
@@ -0,0 +1,141 @@
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.folderwatcher.internal.common.WatcherCommon;
+import org.openhab.binding.folderwatcher.internal.config.FolderLocalWatcherConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link FolderLocalWatcherHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Alexandr Salamatov - Initial contribution
+ */
+@NonNullByDefault
+public class FolderLocalWatcherHandler extends BaseThingHandler {
+    private final Logger logger = LoggerFactory.getLogger(FolderLocalWatcherHandler.class);
+    private @Nullable FolderLocalWatcherConfiguration config;
+    private @Nullable File currentLocalListingFile;
+    private @Nullable ScheduledFuture<?> executionJob;
+    private List<String> currentLocalListing = new ArrayList<>();
+    private List<String> previousLocalListing = new ArrayList<>();
+
+    public FolderLocalWatcherHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(FolderLocalWatcherConfiguration.class);
+        updateStatus(ThingStatus.UNKNOWN);
+
+        currentLocalListingFile = new File(ConfigConstants.getUserDataFolder() + File.separator + "FolderWatcher"
+                + File.separator + thing.getUID().getAsString().replace(':', '_') + ".data");
+        try {
+            previousLocalListing = WatcherCommon.initStorage(currentLocalListingFile);
+        } catch (IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+            logger.debug("Can't write file {}: {}", currentLocalListingFile, e.getMessage());
+            return;
+        }
+
+        if (config.pollIntervalLocal > 0) {
+            updateStatus(ThingStatus.ONLINE);
+            executionJob = scheduler.scheduleWithFixedDelay(this::refreshFolderInformation, config.pollIntervalLocal,
+                    config.pollIntervalLocal, TimeUnit.SECONDS);
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Polling interval can't be null or negative");
+        }
+
+    }
+
+    @Override
+    public void dispose() {
+        if (executionJob != null) {
+            executionJob.cancel(true);
+        }
+    }
+
+    private List<String> listFiles(String path, boolean listHidden, boolean recursive) throws IOException {
+        List<String> dirList = new ArrayList<>();
+        final File dir = new File(path);
+        Optional.ofNullable(dir.listFiles(filter -> listHidden || !filter.isHidden())).map(Arrays::stream)
+                .orElseGet(Stream::empty).forEach(file -> {
```

Comment:
can you please use a regular loop for this?

Reply:
`dir.listFiles` will already return a non-null value so you don't need to check for it.